backend/src/main/java/com/empire/World.java

```diff
@@ -2599,6 +2600,23 @@ public int compare(Node a, Node b) {
 			notifyAll("Harvest Preparations", "People all over the world prepare to reap the harvest upon which they have labored, and hunger is (perhaps only temporarily) banished. All regions will produce their harvest this turn (before eating).");
 		}
 
+		// Check for game end.
+		if (date >= 27 && (date - 27) % 6 == 0) {
+			int votesToEnd = 0;
+			int votesToExtend = 0;
+			for (String kingdom : kingdoms.keySet()) {
+				if ("end".equals(orders.get(kingdom).getOrDefault("end_vote", "end"))) votesToEnd++;
+				else votesToExtend++;
+			}
+			if (votesToEnd * 3 > votesToExtend * 2) {
```

Comment:
Perhaps change your counting scheme to votesToEnd and totalVotes?  While mathematically equivalent to requiring a 2/3 vote to extend, this requires solving a math problem for a reader of this code to determine what the threshold is.  However its counted, variables on one side and constants on the other would help.

Reply:
Done. I tend to avoid this by habit because I don't like to use floating points where it can be avoided, but I agree it degraded readability.

Reply:
I see you've changed it now but just as a correction to my own previous words, this kind of ratio-based check actually results in a 60% majority requirement, the factors for 2/3 majority would have needed to be 2 and 1 respectively in this case.